base/pkg/dir.jl

```diff
@@ -5,7 +5,7 @@ module Dir
 import ..Pkg: DEFAULT_META, META_BRANCH, PkgError
 import ...LibGit2, ...LibGit2.with
 
-const DIR_NAME = ".julia"
+const DIR_NAME = @static Sys.isapple() ? "Library/Julia" : ".julia"
```

Comment:
I agree it would be disruptive.  Perhaps a one-time migration could be added? For example, if the `Library/Julia` directory doesn't exist, create it and move any existing v0.7 stuff from `.julia` and print a notice to the user to manually cp/mv anything that's not versioned from `.julia`.